tealiumlibrary/src/main/java/com/tealium/core/Tealium.kt

```diff
@@ -155,21 +156,23 @@ class Tealium private constructor(val key: String, val config: TealiumConfig, pr
             return
         }
 
-        if (dispatch.timestamp == null) {
-            dispatch.timestamp = System.currentTimeMillis()
+        val dispatchCopy = GenericDispatch(dispatch)
+
+        if (dispatchCopy.timestamp == null) {
```

Comment:
Probably better to set this in the initializer of `GenericDispatch`?

Reply:
good shout - have done so in the pr updates.